storagepool.go

```diff
@@ -43,7 +43,7 @@ func NewStoragePoolEx(client *Client, pool *types.StoragePool) *StoragePool {
 }
 
 // CreateStoragePool creates a storage pool
-func (pd *ProtectionDomain) CreateStoragePool(name string, mediaType string) (string, error) {
+func (pd *ProtectionDomain) CreateStoragePool(name string, mediaType string, optionalPayload map[string]interface{}) (string, error) {
```

Comment:
Instead of accepting a name, mediatype and a random map, why not just accept a types.StoragePoolParam struct?